### ISSUE TITLE
Stuck looping on: Task 'SetupTestMachineForUiTests' on machine '' is taking time. Please Wait

### ISSUE THREAD

**Issue by d-haxton** [Microsoft/vsts-tasks#3962]
Step: `Visual Studio Test Agent Deployment`, have tried using both VS2017 and VS2015 to no different results
Logs from `C:\DtaLogs\DTAExecutionHost.exe`

```
Test User vsuitest

Test UserDomain .

Executing command : powercfg.exe /Change monitor-timeout-ac 0 ; powercfg.exe /Change monitor-timeout-dc 0 

Error :  

Output :  

Currently test user is not logged in. Rebooting machine.

Enabling auto logon
```

Same message repeats for 7000 lines.

Everything was working perfectly fine until one day it wasn't, nothing has changed for the build definition. And, yes, the user is 100% logged in.

**Comment by tanvi-soni** [comment]
The test agent checks for active session for the user. If it does not find one, it reboots. Can you try logging into the test agent machine with user vsuitest and keep the session active. Check if the reboot is happening or if the task passes. 
Additionally please send the specified details to devops_tools@microsoft.com :
i. Version of task, testagent and tfs that you are using. To confirm, you tried installing test agents for VS 2015 as well as for VS 2017.   
ii. Dtalogs and build logs

**Comment by d-haxton** [comment]
Bad email address.

```
The groups only accept messages from people in their organization or on their allowed senders list, and your email address isn't on the lists.
```

**Comment by nigurr** [comment]
@d-haxton,

Sorry for the trouble. This issue is hot fixed in VSTS. https://github.com/Microsoft/vsts-tasks/pull/3985
You can RDP into the test machine and delete this folder contents (C:\Windows\DtlDownloads) and run your build definition again with v2 version selected.
It should work without issues. Please let us know if you are still facing the issue.

**Comment by acesiddhu** [comment]
Closing the issue. Let us know if u face any issues

**Comment by rikkitherockett** [comment]
+1

I've got the exact same problem. What's going on in SetupTestMachineForUiTests? The machine/agent seems to be stuck in restarting its box - no ending.

I cannot cancel the build, I will have to restart the vsts-agent, to explicitly fail the build, and ensure the VS deployment blah blah.. service, which is dynamically registered is disabled once the box is back up, before it restarts again..

**Comment by rikkitherockett** [comment]
Found the issue.

Turns out the Test Agent User - used to configure the agent must be logged in (have an active RDP session), otherwise the PS script SetupTestMachineForUiTests tries to reboot the machine and login using that credential..

In my case - where we have an AD environment (the box is joined to a domain..) there's legal notice upon logging in, and auto-logon (which I was lazy to find a doc about how to enable it...) seems to be disabled..

Therefore no script is able to login using that credentials - even SetupTestMachineForUiTests.ps1.

What I did was to manually login to the box (RDP) using the Test Agent account - and leave the session open. (disconnect rather than logoff..)

These said - that would be nice to have more logging/doc around this part of the code - so that folks don't spend time trying to figure it out. I'm all in for help! :)

Sorry for the spam,
Thanks.
